addon/utils/selection.js

```diff
@@ -63,24 +62,27 @@ export default {
    * @param {Function} itemComparator - comparator for items
    */
   /* eslint-disable complexity */
-  range (items, selectedItems, item, rangeState, itemComparator) {
+  range (items, selectedItems, item, rangeState, itemComparator, itemKey) {
     // If an anchor isn't set, then set the anchor and exit
     const rangeAnchor = rangeState['anchor']
-    if (isNone(rangeAnchor)) {
```

Comment:
I've checked the behaviour of the list selection prior to your last set of comparator changes and the behaviour is correct - what case are you addressing here that requires a change to the logic / flow?